### DISCUSSION TITLE
Vacabulary in writing practice

### DISCUSSION THREAD

**Discussion by sl08154711** [syt0r/Kanji-Dojo#95] [General]
I don't know whether it is a bug or a feature: since version 1.9 the words displayed in writing practice changed. Prior to version 1.9 the words displayed were mostly short (two kanji) and somewhat useful. Now, I see lots of long compunds with katakana or words with four or more kanji which I would find hard to use in a (normal) sentence. Extreme example would be:

![Screenshot_20240201-200309](https://github.com/syt0r/Kanji-Dojo/assets/153142614/6c078807-1f6c-43b4-aac6-9e0985bd6f2d)

I mean, it's one thing to name a thing very precisely but for 朝鮮民主主義人民共和国 I have the feeling I know the whole country's history since the dawn of time after reading that word. Anyway, I don't feel that this is the best example for a word containing 義. There are plenty of other words which would not take so much space and thus would allow to display a few more words.

I like rather short words for the examples displayed during writing practice. Ideally words in which the kanji in question makes up a large percentage of the characters used. Or even better: one word for every possible reading of the kanji and other (common) words after that. This might get tricky for 生 and other kanji with a lot of different readings.

For the discussion:
- Is the change intentional or a consequence of some database changes?
- What do you think about which words should be displayed?
- Maybe a random selection of words that changes every time?

**Comment by syt0r** [comment]
Omg, that's a really huge word, I wonder why it's so high in a list. I think latest database version prioritizes words based on their usage frequency values from jmdict dictionary, the same dictionary that provides words. 

In previous version I was using another data source for frequency, the one mentioned in credits and honestly I didn't notice the difference that much, I only know that for a single-character words it couldn't get frequencies right because there was no context unlike the entries from dictionary, like there could be a pretty uncommon kanji reading or meaning and it would appear at the top top of the list so I had to filter out single-character words

Your idea about random words order is pretty cool, maybe it can be made an optional configuration and it will randomize words up to a certain frequency to avoid uncommon usages
 
I wish there was a better frequency data source, but at least now I made Kanji-Dojo-Data repo with stable data that is not re-generated each database version so it's possible to manually adjust word frequency. I'm also thinking about adding in-app issue report buttons all over the app to hopefully catch issues like this

**Comment by ghost** [comment]


**Reply by syt0r** [reply]
It's possible, although database grew in size a little, I'm probably using updated dictionary and new filtering logic so some expression could have disappeared, it can be investigated if there's a lot less examples now
